db-service/lib/cqn4sql.js

```diff
@@ -430,10 +430,11 @@ function cqn4sql(originalQuery, model = cds.context?.model || cds.model) {
 
       if (col.$refLinks.some(link => link.definition._target?.['@cds.persistence.skip'] === true)) return
 
+      const getName = col => col?.as || col?.ref?.at?.(-1)
       const flatColumns = getFlatColumnsFor(col, { baseName, columnAlias, tableAlias })
       flatColumns.forEach(flatColumn => {
-        const { as } = flatColumn
-        if (!(as && transformedColumns.some(inserted => inserted?.as === as))) transformedColumns.push(flatColumn)
+        const name = getName(flatColumn)
+        if (!transformedColumns.some(inserted => getName(inserted) === name)) transformedColumns.push(flatColumn)
```

Comment:
We should make `transformedColumns` an object instead of an array. Then add to it by the like of:
```suggestion
        if (!(name in transformedColumns)) transformedColumns[name] = flatColumn
```
But after all: can we really just silently skip conflicting column names? 
What about that?: 
```sql
SELECT foo, bar as foo from Boo;
```
```

Reply:
I am not certain whether this behavior is ok or not, but my expectation is that the `foo, bar as foo` example would be caught by infer [here](https://github.com/cap-js/cds-dbs/blob/main/db-service/lib/infer/index.js#L292).

Reply:
If we let
```
SELECT DraftAdministrativeData, DraftAdministrativeData_DraftUUID, TravelUUID FROM TravelService.Travel.drafts
```
pass then the result data must be present both in the structure `DraftAdministrativeData` and in the plain field `DraftAdministrativeData_DraftUUID`
I think simply removing one of them is not an option.

Reply:
Currently `DraftAdministrativeData` turns into `DraftAdministrativeData_DraftUUID` which would indicate that there is no difference. If `DraftAdministrativeData` was an expand it would be different.

Reply:
Additionally in the test I added it is required to compile the model for `OData` as then the foreign keys can be selected. Where if the model is not recompiled the foreign key columns are not present in the model and can not be selected. So this might just be a side effect from the decision that it is allowed to explicitly select generated foreign key columns.